actors/evm/src/lib.rs

```diff
@@ -312,6 +313,80 @@ fn handle_filecoin_method_input(method: u64, codec: u64, params: &[u8]) -> Vec<u
     buf
 }
 
+/// Decode the response from "filecoin_native_method". We expect:
+///
+/// 1. The exit code (u32).
+/// 2. The codec (u64).
+/// 3. The data (bytes).
+///
+/// According to the solidity ABI.
+fn handle_filecoin_method_output(mut output: Vec<u8>) -> Result<Option<IpldBlock>, ActorError> {
+    // Short-circuit if empty.
+    if output.is_empty() {
+        return Ok(None);
+    }
+
+    // TODO: This should use the "parameter" reader. But we need to move it and likely refactor
+    // something.
```

Comment:
This is a pretty big TODO, dont want to trip on the same things we did in precompiles

Reply:
I agree. I can try to extract it out now.